### PR TITLE
Add substring remover mapper

### PR DESCRIPTION
**Pull request by cakiki**
This function is meant to strip repeated strings like the ones here: https://github.com/bigscience-workshop/catalogue_data/issues/5#issuecomment-1057424610

### CODE REVIEW

**Comment by lvwerra** [comment]
Can you add your function to the following two places?
1. the `__init__.py` in the same folder
2. import it to `clean.py` and add it to the dictionary

**Review by thomasw21 — COMMENTED** [review]
I don't know the full context, but modifying substrings is risky business IMO. IMO the safer is just to drop lines that have the a substring you want to remove.

File: clean_helpers/map_strip_substring.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+def build_substring_stripper(list_of_substrings):
```
[the remark this answers is not in the document]
Comment (cakiki) [reply clean_helpers/map_strip_substring.py:1]: This is mostly for the non-wikipedia wikis. Dropping lines means dropping entire documents in this case as a lot of them have no `\n`

File: clean_helpers/map_strip_substring.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    def strip_substrings(batch):
+        text_batch = batch["text"]
+        for substring in list_of_substrings:
+            text_batch = [text.replace(substring, "") for text in text_batch]
```
[the remark this answers is not in the document]
Comment (lvwerra) [reply clean_helpers/map_strip_substring.py:5]: I think the most of the datasets we apply this to are smallish.

[the remark this answers is not in the document]
Comment (thomasw21) [reply clean_helpers/map_strip_substring.py:5]: Yeah I'm pretty sure we can build a regex dynamically. Something along the lines of https://stackoverflow.com/a/6117124 The worry I have, is `replace` in a loop creates as many strings are you have elt in your list for substrings which I'm pretty sure is VERY slow.

Though if you think it's going to be quite fast overall as the datasets are small, let's not care.

[the remark this answers is not in the document]
Comment (cakiki) [reply clean_helpers/map_strip_substring.py:5]: They are small and few in number.

File: clean.py [hunk 3]
```diff
@@ -17,7 +18,8 @@
 # Map functions: function(batch: Dict) -> Dict
 MAPS = {
     "replace_newline_with_space": replace_newline_with_space,
-    "remove_lines_with_code": remove_lines_with_code
+    "remove_lines_with_code": remove_lines_with_code,
+    "strip_substrings_kn_wikisource": build_substring_stripper([["അവലംബം","പുറത്തേക്കുള്ള കണ്ണികൾ"]]),
```
Comment (lvwerra) [inline clean.py:22]: These seem to be "Source", "External links". As mentioned the slack thread I think they belong to the document and we should only remove them if they don't occur in a natural way.

File: clean_helpers/map_strip_substring.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        return {
+            **batch,
+            "text": text_batch
+        }
+    return strip_substrings
+
+
+en_wiktionary_stripper = build_substring_stripper(
```
Comment (lvwerra) [inline clean_helpers/map_strip_substring.py:13]: you should add that to the `__init__` and `clean.py` as well.

**Review by lvwerra — COMMENTED** [review]
I think this is almost there. Let me know if you need help resolving the conflicts.

**Review by lvwerra — APPROVED** [review]

**Pull request merged**